src/main/scala/com/fulcrumgenomics/bam/SamRecordClipper.scala

```diff
@@ -354,9 +354,14 @@ class SamRecordClipper(val mode: ClippingMode, val autoClipAttributes: Boolean)
     * @return the additional number of bases clipped (3' end in sequencing order) for the read and mate respectively
     */
   def clipExtendingPastMateEnds(rec: SamRecord, mate: SamRecord): (Int, Int) = {
-    val basesClipped1 = clipExtendingPastMateEnd(rec=rec, mateUnclippedStart=mate.unclippedStart, mateUnclippedEnd=mate.unclippedEnd)
-    val basesClipped2 = clipExtendingPastMateEnd(rec=mate, mateUnclippedStart=rec.unclippedStart, mateUnclippedEnd=rec.unclippedEnd)
-    (basesClipped1, basesClipped2)
+    if (rec.isFrPair) {
+      val basesClipped1 = clipExtendingPastMateEnd(rec=rec, mateUnclippedStart=mate.unclippedStart, mateUnclippedEnd=mate.unclippedEnd)
+      val basesClipped2 = clipExtendingPastMateEnd(rec=mate, mateUnclippedStart=rec.unclippedStart, mateUnclippedEnd=rec.unclippedEnd)
+      (basesClipped1, basesClipped2)
+    }
+    else {
+      (0, 0)
+    }
```

Comment:
The two calls to `clipExtendingPastMateEnd` already performed this guard and returned 0 each. Are you doing this so that we don't have to spend time computing the unclipped end positions if we don't need them?